app/controllers/authenticate_controller.rb

```diff
@@ -100,8 +120,24 @@ def k8s_inject_client_cert
 
   private
 
+  def handle_login_error(err)
+    logger.warn("Authentication Error: #{err.inspect}")
+    err.backtrace.each do |line|
+      logger.debug(line)
+    end
+
+    case err
+    when Errors::Authentication::Security::NotWhitelisted,
+      Errors::Authentication::Security::ServiceNotDefined,
+      Errors::Authentication::Security::AccountNotDefined
+      raise Unauthorized
+    else
+      raise err
+    end
+  end
+
   def handle_authentication_error(err)
-    logger.debug("Authentication Error: #{err.inspect}")
+    logger.warn("Authentication Error: #{err.inspect}")
```

Comment:
i don't really like this happening under the hood. Are we sure the err doesn't contain any sensitive information?
@shaharglazner do you see any concern here?

Reply:
@sgnn7 can you weigh in? I remember talking to you about this after the hackathon

Reply:
@orenbm yeah - this action is unknown if it will leak any credentials. we'd need to investigate the matrix of possible errors and authenticators to confirm that this won't dump stuff into the logs that shouldn't be there :/

Reply:
@jtuttle please change this to `debug` and create another issue to change this to `warn`. let's not delay this PR from being merged because of this. same goes for the warn in `handle_login_error`. thanks!